### ISSUE TITLE
[BUG] ReportPortalClientModule incorrect check for a null value 

### ISSUE THREAD

**Issue by shaburov** [reportportal/client-java#9]
client-java-3.0.7

Incorrect instantiation of the CloseableHttpAsyncClient with a non-self-signed certificate.
PropertiesLoader injected with a Properties object, in which the values are converted to a string (String.valueOf())

![image](https://user-images.githubusercontent.com/15269131/32146055-f73b508a-bce2-11e7-8846-fe6264d26ffd.png)

![image](https://user-images.githubusercontent.com/15269131/32147150-3e9bb4e6-bcf3-11e7-9491-6a042a9ede06.png)

```
Caused by: com.epam.reportportal.exception.InternalReportPortalClientException: Unable to load trust store
	at com.epam.reportportal.guice.ReportPortalClientModule.provideHttpClient(ReportPortalClientModule.java:153)
	at com.epam.reportportal.guice.ReportPortalClientModule$$FastClassByGuice$$20c74f8c.invoke(<generated>)
	at rp.com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:264)
	at rp.com.google.inject.internal.ProviderMethod$Factory.provision(ProviderMethod.java:401)
	at rp.com.google.inject.internal.ProviderMethod$Factory.get(ProviderMethod.java:376)
	at rp.com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at rp.com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at rp.com.google.inject.internal.ProviderMethod$Factory.provision(ProviderMethod.java:402)
	at rp.com.google.inject.internal.ProviderMethod$Factory.get(ProviderMethod.java:376)
	at rp.com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at rp.com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at rp.com.google.inject.internal.ProviderMethod$Factory.provision(ProviderMethod.java:402)
	at rp.com.google.inject.internal.ProviderMethod$Factory.get(ProviderMethod.java:376)
	at rp.com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at rp.com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at rp.com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at rp.com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at rp.com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at rp.com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at rp.com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at rp.com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:120)
	at rp.com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
	at rp.com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:268)
	at rp.com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:61)
	at rp.com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at rp.com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at rp.com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at rp.com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at rp.com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at rp.com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1019)
	at rp.com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085)
	at rp.com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1015)
	... 10 more
```

**Comment by avarabyeu** [comment]
Fixes in 4.0

**Comment by sudheersinghcontinuum** [comment]
I am getting this same error, and I am using 4.0-

2020-04-01 13:24:34 ERROR ReportPortal:240 - Cannot build ReportPortal client
com.epam.reportportal.exception.InternalReportPortalClientException: Unable to load trust store
	at com.epam.reportportal.service.ReportPortal$Builder.defaultClient(ReportPortal.java:286)
	at com.epam.reportportal.service.ReportPortal$Builder.build(ReportPortal.java:233)
	at com.epam.reportportal.cucumber.AbstractReporter.buildReportPortal(AbstractReporter.java:107)
	at com.epam.reportportal.cucumber.AbstractReporter$1.get(AbstractReporter.java:77)
	at com.epam.reportportal.cucumber.AbstractReporter$1.get(AbstractReporter.java:70)
	at rp.com.google.common.base.Suppliers$MemoizingSupplier.get(Suppliers.java:120)
	at com.epam.reportportal.cucumber.ScenarioReporter$1.get(ScenarioReporter.java:62)
	at com.epam.reportportal.cucumber.ScenarioReporter$1.get(ScenarioReporter.java:55)
	at rp.com.google.common.base.Suppliers$MemoizingSupplier.get(Suppliers.java:120)
	at com.epam.reportportal.cucumber.ScenarioReporter.getRootItemId(ScenarioReporter.java:106)
	at com.epam.reportportal.cucumber.AbstractReporter.beforeFeature(AbstractReporter.java:127)
	at com.epam.reportportal.cucumber.AbstractReporter.feature(AbstractReporter.java:309)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at cucumber.runtime.Utils$1.call(Utils.java:40)
	at cucumber.runtime.Timeout.timeout(Timeout.java:16)
	at cucumber.runtime.Utils.invoke(Utils.java:34)
	at cucumber.runtime.RuntimeOptions$1.invoke(RuntimeOptions.java:294)
	at com.sun.proxy.$Proxy18.feature(Unknown Source)
	at cucumber.runtime.model.CucumberFeature.run(CucumberFeature.java:161)
	at cucumber.api.testng.TestNGCucumberRunner.runCucumber(TestNGCucumberRunner.java:63)
	at continuum.cucumber.testrunner.TestRunner.feature(TestRunner.java:49)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2020-04-01 13:24:34 ERROR ReportPortal:240 - Cannot build ReportPortal client
com.epam.reportportal.exception.InternalReportPortalClientException: Unable to load trust store
	at com.epam.reportportal.service.ReportPortal$Builder.defaultClient(ReportPortal.java:286)
	at com.epam.reportportal.service.ReportPortal$Builder.build(ReportPortal.java:233)
	at com.epam.reportportal.cucumber.AbstractReporter.buildReportPortal(AbstractReporter.java:107)
	at com.epam.reportportal.cucumber.AbstractReporter$1.get(AbstractReporter.java:77)
	at com.epam.reportportal.cucumber.AbstractReporter$1.get(AbstractReporter.java:70)
	at rp.com.google.common.base.Suppliers$MemoizingSupplier.get(Suppliers.java:120)
	at com.epam.reportportal.cucumber.ScenarioReporter.afterLaunch(ScenarioReporter.java:111)
	at com.epam.reportportal.cucumber.AbstractReporter.close(AbstractReporter.java:372)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at cucumber.runtime.Utils$1.call(Utils.java:40)
	at cucumber.runtime.Timeout.timeout(Timeout.java:16)
	at cucumber.runtime.Utils.invoke(Utils.java:34)
	at cucumber.runtime.RuntimeOptions$1.invoke(RuntimeOptions.java:294)
	at com.sun.proxy.$Proxy18.close(Unknown Source)
	at cucumber.api.testng.TestNGCucumberRunner.finish(TestNGCucumberRunner.java:77)
	at continuum.cucumber.testrunner.TestRunner.tearDownClass(TestRunner.java:64)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:523)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:224)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeAfterClassMethods(TestMethodWorker.java:212)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 2.526 s <<< FAILURE! - in TestSuite
[ERROR] feature(continuum.cucumber.testrunner.TestRunner)  Time elapsed: 0.595 s  <<< FAILURE!
com.epam.reportportal.exception.InternalReportPortalClientException: Cannot build ReportPortal client
	at continuum.cucumber.testrunner.TestRunner.feature(TestRunner.java:49)
Caused by: com.epam.reportportal.exception.InternalReportPortalClientException: Unable to load trust store
	at continuum.cucumber.testrunner.TestRunner.feature(TestRunner.java:49)
